BDArmory/UI/BDAEditorArmorWindow.cs

```diff
@@ -1123,6 +1175,290 @@ void Visualize()
+                                        {
+                                            break;
+                                        }
+                                        ModuleCommand AIParent = null;
+                                        if (kvp.Value[0].parent) AIParent = kvp.Value[0].parent.FindModuleImplementing<ModuleCommand>();
+                                        if (AIParent == null)
+                                        {
+                                            nonCockpitAI = true;
+                                        }
+                                        break;
+                                    }
+                                case "ModuleCommand":
+                                case "KerbalSeat":
+                                    {
+                                        int crewCount = kvp.Value[0].FindModuleImplementing<ModuleCommand>().minimumCrew;
+                                        if (crewCount <= 0)
+                                        {
+                                            if (!string.IsNullOrEmpty(blacklistedParts)) blacklistedParts += " | ";
+                                            blacklistedParts += $"{kvp.Value[0].partInfo.title}(Probecore: {kvp.Value.Count}/0)";
+                                        }
+                                        /*
+                                        if (kvp.Value[0] != EditorLogic.fetch.ship.Parts[0])
+                                        {
+                                            nonRootCockpit = true;
+                                        }
+                                        */
+                                        break;
+                                    }
+                            }
+                        }
+                    }
+                    if (BDArmorySettings.MAX_PWING_LIFT > 0)
+                    {
+                        foreach (var part in EditorLogic.fetch.ship.Parts) //not ideal, but this needs to fire onVesselModified, not onPartPlaced, but linking this to Visualizer's parts eval only updates when that does
+                        {
+                            if (part.name.Contains("B9.Aero.Wing.Procedural.Type"))
+                            {
+                                ModuleLiftingSurface wing = part.GetComponent<ModuleLiftingSurface>();
+                                if (wing != null && wing.deflectionLiftCoeff > 0f)
+                                {
+                                    if (wing.deflectionLiftCoeff > BDArmorySettings.MAX_PWING_LIFT)
+                                        oversizedPWings++;
+                                }
+                            }
+                        }
+                    }
+                }
+                StringBuilder evaluationstring = new StringBuilder();
+                ScreenMessage vessellegality = new ScreenMessage("", 7.0f, ScreenMessageStyle.LOWER_CENTER);
```

Comment:
Moving this outside of the function allows the message to be cleared properly. It's not being cleared on `ScreenMessages.RemoveMessage(vessellegality);` below currently as that's a different `ScreenMessage` instance.